tests/testthat/helper-mocks.R

```diff
@@ -0,0 +1,52 @@
+mock_cran_version <- function(version, .env = caller_env()) {
+  local_mocked_bindings(cran_version = function() version, .env = .env)
+}
+
+mock_check_installed <- function(.env = caller_env()) {
+  local_mocked_bindings(check_installed = function(...) TRUE, .env = .env)
+}
+
+mock_check_is_package <- function(.env = caller_env()) {
+  local_mocked_bindings(
+    check_is_package = function(...) invisible(),
+    .env = .env
+  )
+}
+
+mock_rstudio_not_available <- function(.env = caller_env()) {
```

Comment:
This is the only helper that isn't of the pattern `mock_[function_name]()`. I think having them all fit that pattern helps the reader understand exactly what function is being mocked. What about `mock_rstudio_available_false()`? Are there any cases where it's mocked to be `TRUE`? It could then be:

```r
mock_rstudio_available <- function(is_available) {
  local_mocked_bindings(
    rstudio_available = function(...) is_available, 
    .env = .env
  )
}
```

Reply:
Yeah it started as `mock_rstudio_available()` and it made `rstudio_available()` always return `FALSE`.

Then I thought I should expose the hard-wired value as the argument (`FALSE`, not available, vs. `TRUE` available).

But then I realized we _never_ needed the `TRUE` version.

So then I settled on `mock_rstudio_not_available()`.

Reply:
That makes sense, it just stood out to me as different - `mock_check_installed()`, `mock_ui_yeah()` etc all have a hard-wired return value but don't signal it in the function name. 

Maybe the difference is that that some mockers are used to breeze by tricky bits (like UI menus, checking github repos etc) so you always want them to return a value that allows the code to run, but some might to be emulate different environments?